### PR TITLE
Docs: adding document lists feature to the guide.

### LINKED ISSUE #11705: [Advanced lists] add guide to CKE5 documentation
Add the advanced list information and demo.

Based on talk in the core team, this should be added to the existing lists guide to ultimately replace the old feature once it's extinguished.

### PR DESCRIPTION
**Pull request by godai78**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Type: Adding document lists (advanced lists) to the docs lists guide. Closes #11705.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by oleq** [comment]
I added the demo content for `DocumentLists`. I used images from unsplash:

*   [https://unsplash.com/photos/ON5qDh2m-Ro](https://unsplash.com/photos/ON5qDh2m-Ro)
*   [https://unsplash.com/photos/7xiADv3VZ0k](https://unsplash.com/photos/7xiADv3VZ0k)
*   [https://unsplash.com/photos/3OiYMgDKJ6k](https://unsplash.com/photos/3OiYMgDKJ6k)

so we're probably cool with copyrights but we need to double check if authors must be credited somewhere.

**Comment by godai78** [comment]
> so we're probably cool with copyrights but we need to double check if authors must be credited somewhere.

https://unsplash.com/license:

*   **No permission needed** (though attribution is appreciated!)

We can decide on our own, then.

File: packages/ckeditor5-list/docs/_snippets/features/lists-document.html [hunk 1]
```diff
@@ -6,39 +6,39 @@ <h3>...and stay on the budget.</h3>
...
+				<figcaption>Ocean shore of Gran Canaria</figcaption>
 			</figure>
-			<p>Gran Canaria is a popular tourist destination. The island is often called "a miniature continent". This is owed to the fact, that is offers a great variety of climates and landscapes, from golden, sandy beaches, through rocky ravines, white dunes and charming villages. In the mid-2010s over 3.5 million tourists visited Gran Canaria annually.</p>
+			<p>Gran Canaria is a popular tourist destination. The island is often called "a miniature continent". This is owed to the fact, that it offers a great variety of climates and landscapes, from golden, sandy beaches, through rocky ravines, white dunes, and charming villages. In the mid-2010s over 3.5 million tourists visited Gran Canaria annually.</p>
 		</li>
 		<li>
-			<h4>Florence, Italy</h4>
-			<p>Florence, established in 59BC by Julius Caesar himself got its name from two river that surrounded the settlement. Thanks to centuries of history, the capital of Tuscany has plenty to offer for tourists hungry of sightseeing. Some of the most interesting places to visit include:</p>
+			<p>Florence, Italy</p>
```
Comment (oleq) [inline packages/ckeditor5-list/docs/_snippets/features/lists-document.html:14]: Why is this a paragraph now?

Comment (godai78) [reply packages/ckeditor5-list/docs/_snippets/features/lists-document.html:14]: Not intentionally, must've changed during some paste. I'll restore it.

**Pull request closed without merging**